### PR TITLE
[WIP] Read Iceberg to DeltacatDataset

### PR DESCRIPTION
**Pull request by JonasJ-ap**
Adds a `read_iceberg` read API, which uses [`pyiceberg`](https://github.com/apache/iceberg/tree/master/python/pyiceberg) and Ray Data API `read_parquet`.

Waiting for `pyiceberg0.4.0" Release (Release Vote Begins)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def find_version(*paths):
         "pyarrow == 10.0.1",
         "pydantic == 1.10.4",
         "ray[default] ~= 2.0",
-        "s3fs == 2022.2.0",
```
Comment (JonasJ-ap) [inline setup.py:44]: `pyiceberg 0.3.0` requires `2023.1.0` in its specification. The version range will be widen in the next release (0.4.0)

**Review by pdames — CHANGES_REQUESTED** [review]
As a high-level comment, let's (1) target merging this into a new `iceberg` branch instead of the `main` branch for now and (2) ensure that all changes pass the linter by following the steps  at https://github.com/ray-project/deltacat/blob/main/README-development.md. You can also review the complete set of suggested changes here that are passing `make test-integration` on my Macbook at https://github.com/pdames/deltacat/tree/add_read_iceberg.

**Review by JonasJ-ap — COMMENTED** [review]
@pdames Thank you for your review and suggestions. I've incorporated your suggested update and revised the integration test structure.

**Comment by pdames** [comment]
LGTM - nice work @JonasJ-ap!

**Pull request merged**